test/index.js

```diff
@@ -183,3 +183,171 @@ describe('metalsmith-serve with custom indexFile', function(){
 
 });
 
+
+// not_found file serving and redirects
+describe('metalsmith-serve', function() {
+
+  var metalsmith;
+  var servePlugin;
+
+  before(function(done) {
+    metalsmith = Metalsmith("test/fixtures/site");
+
+    servePlugin = serve({
+      verbose: false,
+      "port": port,
+      "not_found": "/404.html",
```

Comment:
Yes, I like the idea however I can't imagine much demand for other responses with the purpose of the module being to serve static content.

The 404 compared to the other codes will be slightly inconsistent as the 404 is a path to a file to serve and the rest will be redirects etc.
e.g.
```js
{
  verbose: false,
  ...
  http_codes: {
    404: "/404.html",
    301: {
      "/old_path/": "/new_path/",
      ...
    },
    ...
}